### PR TITLE
NOD | Confirmation page update

### PR DESCRIPTION
**Pull request by Mottie**
## Summary

- _(Summarize the changes that have been made to the platform)_
  > Adding Notice of Disagreement confirmation page with PDF download link & form data. Note this work is behind a feature flag and the Lighthouse PDF download API is not currently available
- _(If bug, how to reproduce)_
- _(What is the solution, why is this the solution)_
  > Confirmation page changes based on user research
- _(Which team do you work for, does your team own the maintenance of this component?)_
  > Benefits Decision Reviews
- _(If using a flipper, what is the end date of the flipper being required/success criteria being targeted)_

## Related issue(s)

- [#81507](https://github.com/department-of-veterans-affairs/va.gov-team/issues/81507)
- [Design](https://www.figma.com/file/BplQkEDZzD8NPPNmLrhvXv/Request-a-Board-Appeal-%2F-Notice-of-Disagreement-(VA-10182)?type=design&node-id=831-76228&mode=dev)

## Testing done

- _Describe what the old behavior was prior to the change_
  > Preliminary work; tests to be added in a separate PR
- _Describe the steps required to verify your changes are working as expected_
  > Fill out NOD form & submit in staging - PDF download link currently points to a blank NOD PDF 
- _Describe the tests completed and the results_
- _Exclusively stating 'Specs and automated tests passing' is NOT acceptable as appropriate testing
- _Optionally, provide a link to your [test plan](https://depo-platform-documentation.scrollhelp.site/developer-docs/create-a-test-plan-in-testrail) and [test execution records](https://depo-platform-documentation.scrollhelp.site/developer-docs/execute-tests-in-testrail)_

## Screenshots

|         |  |
| ------ | ----- |
| Submission with evidence | full page screenshot of NOD confirmation page with evidence uploads |
| Submission with hearing choice | full page screenshot of NOD confirmation page with hearing choice selected |
| PDF generating | pdf generating loading indicator |
| PDF generation taking more than 1 minute | loading indicator with warning alert regarding the process taking longer than expected |
| PDF generation error | PDF generation error message |
| PDF generation success | Link to download generated PDF showing after successful generation |

## What areas of the site does it impact?

Notice of Disagreement (behind a feature flag)

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [x] Screenshot of the developed feature is added
- [x] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/applications/appeals/10182/components/ConfirmationPageV2.jsx [hunk 1]
```diff
@@ -0,0 +1,206 @@
+/* eslint-disable jsx-a11y/no-redundant-roles */
```
[the remark this answers is not in the document]
Comment (Mottie) [reply src/applications/appeals/10182/components/ConfirmationPageV2.jsx:1]: Adding a `role="list"` to `ul` with `list-style: none` to work around a problem with Safari not treating the `ul` as a list - recommended by @davidakennedy

File: src/applications/appeals/shared/components/ConfirmationIssues.jsx [hunk 2]
```diff
@@ -0,0 +1,81 @@
+/* eslint-disable jsx-a11y/no-redundant-roles */
```
[the remark this answers is not in the document]
Comment (Mottie) [reply src/applications/appeals/shared/components/ConfirmationIssues.jsx:1]: Adding a `role="list"` to `ul` with `list-style: none` to work around a problem with Safari not treating the `ul` as a list.

File: src/applications/appeals/shared/components/ConfirmationPdfMessages.jsx [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+            setApiState('successful');
+          } else {
+            setApiState('error');
+          }
+        })
+        .catch(() => {
+          setApiState('error');
+        });
+    },
+    [], // eslint-disable-line react-hooks/exhaustive-deps
```
[the remark this answers is not in the document]
Comment (Mottie) [reply src/applications/appeals/shared/components/ConfirmationPdfMessages.jsx:42]: Only run code inside `useEffect` once.

File: src/applications/appeals/shared/components/ConfirmationPersonalInfo.jsx [hunk 4]
```diff
@@ -0,0 +1,139 @@
+/* eslint-disable jsx-a11y/no-redundant-roles */
```
[the remark this answers is not in the document]
Comment (Mottie) [reply src/applications/appeals/shared/components/ConfirmationPersonalInfo.jsx:1]: Adding a `role="list"` to `ul` with `list-style: none` to work around a problem with Safari not treating the `ul` as a list.

**Review by anniebtran — APPROVED** [review]
some small suggestions, not blocking though

**Review by harshilmoradia — APPROVED** [review]

**Pull request merged**